enhancements/user-monitoring.md

```diff
@@ -0,0 +1,248 @@
+
+#### US6
+
+As a service owner, I’d like to query the metrics I have access to from the OpenShift console.
+
+#### US7
+
+As a member of the operations team, I’d like to make sure that one team doesn't have access to metrics from another team.
+
+#### US8
+
+As a member of the operations team, I'd like the ability for application developers to access container cpu and memory usage metrics.
+
+### Implementation Details/Notes/Constraints
+
+#### Isolation
+
+The cluster monitoring as well as the user monitoring stack are composed of custom resources
+reconciled by two separate prometheus operators.
+Both prometheus operators are under supervision of cluster monitoring operator.
+This allows a seperation of concerns in terms of ugpradability, security settings, and scalability.
+The cluster monitoring Prometheus Operator continues to be deployed in the `openshift-monitoring` namespace.
+The user monitoring Prometheus Operator will be deployed in the `openshift-user-monitoring` namespace.
+
+Currently, cluster monitoring operator scans all namespaces filtering openshift namespaces labelled with
+`openshift.io/cluster-monitoring: "true"` and takes this set of namespaces as the _allowed list_ of reconcilable
+namespaces for the existing cluster monitoring prometheus operator.
+
+The same set of openshift namespaces that exist today will be configured as a _deny list_ of reconcilable namespaces
+at the user monitoring prometheus operator.
+Deny listing of namespaces was added recently in prometheus operator, see [3].
+
+To make sure that users cannot add Prometheus instances in user namespaces,
+a new WIP feature [4] will ensure that reconcilation of Prometheus custom resources
+will be limited to the `openshift-user-monitoring` namespace for the user monitoring prometheus operator
+and `openshift-monitoring` namespace for the cluster monitoring prometheus operator.
+
+![](user-namespaces-custom-resources.png)
+
+User monitoring Prometheus Operator (green) then will reconcile Prometheus custom resources
+in the `openshift-user-monitoring` namespace only.
+It ignores Prometheus custom resources in any other namespace.
+The same holds true for the cluster monitoring prometheus operator (red).
+
+#### Aggregation
+
+The front facing Kubernetes service has to make it as simple as possible
+for integrators to leverage both user monitoring as well as cluster monitoring.
+
+This is achieved by using Thanos Querier which allows to aggregate
+and optionally deduplicate the cluster and user monitoring metrics backends
+under single Prometheus Query endpoint.
+
+Usage of the Thanos querier component inside cluster monitoring has many advantages:
+1. Run-time deduplication of metrics available today.
+2. Potentially gap-filling of metrics series in the Prometheus HA user and cluster monitoring setup in the future.
```

Comment:
Isn't that the same as point 1? 

Also "in future" relies to cluster mon setup which is 1 replica, but it might sound like everything is in the future (:

Reply:
it is the same technically, but from a use case perspective, i think we can leave them separate (it was at least an eye-opener for me).

Removed "potentially" and "in the future", indeed, it is available today :+1: